### DISCUSSION TITLE
hi what was the endpoint for on_message

### DISCUSSION THREAD

**Discussion by skaaddd** [Rapptz/discord.py#9265] [Q&A]
i want to make a nitro sniper with discord.py but it doesnt support reciveing message, it only recieve it self not other people so i want the api endpoint to do it thanks for replies

**Comment by lightlybakedman** [comment]
discord.py does not support selfbots.

**Reply by skaaddd** [reply]
yes it doesnt but how does on message works

**Reply by IAmTomahawkx** [reply]
The user gateway is different from the bot gateway. Discord.py will not work properly with the user gateway. There is no fixing that. Additionally, you will not receive help from the community with selfbotting

**Reply by skaaddd** [reply]
thank you for ur response

**Reply by skaaddd** [reply]
so if i want to find the bot gateway
which file of the src of discord py is it ?

**Reply by lightlybakedman** [reply]
Why? You can't use the bot gateway as a user
